src/Cake.Common/Tools/NuGet/Pack/NuGetPacker.cs

```diff
@@ -175,6 +176,18 @@ private ProcessArgumentBuilder GetArguments(FilePath filePath, NuGetPackSettings
                 builder.Append(settings.Verbosity.Value.ToString().ToLowerInvariant());
             }
 
+            // Properties
+            if (settings.Properties != null && settings.Properties.Count > 0)
+            {
+                if (settings.Properties.Values.Any(x => x == null))
+                {
+                    throw new CakeException("Properties values can not be null.");
+                }
+                builder.Append("-Properties");
+                builder.Append(string.Join(";",
+                    settings.Properties.Select(property => property.Key + "=" + property.Value)));
```

Comment:
Would we need any escaping here?
Also personally prefer string.Format or string.Contcat instead of adding strings together with + sign.